models/classes/theme/ThemeServiceAbstract.php

```diff
@@ -216,9 +224,17 @@ protected function getThemeDetailsProviders()
 
     protected function isTaoAsToolEnabled(): bool
     {
-        return $this->getServiceManager()
-            ->getContainer()
-            ->get(FeatureFlagChecker::class)
+        return $this->getFeatureFlagChecker()
             ->isEnabled(FeatureFlagCheckerInterface::FEATURE_FLAG_TAO_AS_A_TOOL);
     }
+
+    protected function getFeatureFlagChecker(): FeatureFlagCheckerInterface
```

Comment:
I recommend to apply least privilege principle unless we really need to use it on child classes. Otherwise, we must never use protected, so we can reduce the surface of any breaking change.

```suggestion
    private function getFeatureFlagChecker(): FeatureFlagCheckerInterface
```

Reply:
Correct. And I was sure you would say this 😄 
I decided to use protected instead of private because the other similar methods were doing it, and also because this is an abstract class with method that would be consumed by children